### PR TITLE
Review wanted - todo Frida

### PR DESCRIPTION
**Pull request by fridamaria**
Link to my deployed project: https://fridamaria-todos.netlify.app/

My reflections on how this week's project turned out: it was a fun project and I liked the way it turned out. Good to be able to follow Van's code along step by step as it helped me understand Redux better. This week's videos made it clearer as well, was very confused after the meet up. 

Things I'd like to have clarified or explained in more detail: I think I just need to look at the videos again and do some more reading. Redux is not as intuitive as React at this point, but I will get there :)

### CODE REVIEW

File: code/public/index.html [hunk 1]
```diff
@@ -1,34 +1,22 @@
...
+<head>
+  <meta charset="utf-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <link rel="icon" href="%PUBLIC_URL%/2705.svg">
+  <link href="https://fonts.googleapis.com/css2?family=Playfair+Display:wght@900&display=swap" rel="stylesheet">
+  <meta property="og:url" content="https://fridamaria-todos.netlify.app/" />
+  <meta property="og:type" content="website" />
+  <meta property="og:title" content="Todo App" />
+  <meta property="og:description" content="Small todo app" />
+  <meta property="og:image" content="%PUBLIC_URL%/project-todo.png" />
```
Comment (idlefingers) [inline code/public/index.html:13]: 😎

File: code/src/components/EmptyState.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  margin: 2px 0;
+  font-size: 12px;
+  color: #3e3e2d;
+
+  @media (min-width: 768px) {
+    font-size: 14px;
+  }
+`
+
+export const EmptyState = () => {
```
Comment (idlefingers) [inline code/src/components/EmptyState.js:50]: Nice with something to show when the list is empty! 💯

File: code/src/components/Filters.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  const dispatch = useDispatch()
+
+  //Functions filtering active, done and all todos
+  const showActive = () => dispatch(todos.actions.showActive())
+  const showDone = () => dispatch(todos.actions.showDone())
+  const showAll = () => dispatch(todos.actions.showAll())
+
+  return (
+    <>
+      {itemsLength > 0 &&
```
Comment (idlefingers) [inline code/src/components/Filters.js]: Nitpicking here, but I think that if the component shouldn't render anything in a certain condition, then it's clearer to bail out before you start writing jsx. Something like this:

```jsx
export const Filter = () => {
  // ... all the initial stuff here

  if (itemsLength === 0) {
    return null
  }

  return (
    <FilterWrapper>
      ...
    </FilterWrapper>
  )
}
```

File: code/src/components/Filters.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        <FilterWrapper>
+
+          <Button
+            aria-label='Show active tasks'
+            onClick={showActive}
+            width={'100%'}
+            margin={'5px auto'}
```
Comment (idlefingers) [inline code/src/components/Filters.js]: There's no need for the curly braces when you're returning a string like this. You can just do `width="100%" margin="5px auto"` etc

File: code/src/components/TodoInput.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    <TodoForm onSubmit={handleOnSubmit}>
+
+      <TodoInputField
+        aria-label="Enter task here"
+        type='text'
+        onChange={e => setInputValue(e.target.value)}
+        value={inputValue}
+        placeholder='Add task...'
+        maxLength={50}
+      ></TodoInputField>
```
Comment (idlefingers) [inline code/src/components/TodoInput.js]: When a component (or div or any other element) has no children, you can write it as a self-closing tag:

```jsx
<TodoInputField
  aria-label="Enter task here"
  type='text'
  onChange={e => setInputValue(e.target.value)}
  value={inputValue}
  placeholder='Add task...'
  maxLength={50} />
```

File: code/src/components/TodoItem.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+  //Removes todo from array
+  const onRemoveClicked = () => dispatch(todos.actions.removeTodo({ itemIndex }))
+
+  //Marks todo as done
+  const onCheckClick = () => dispatch(todos.actions.setDone({ itemIndex }))
+
+  return (
+    <>
+      {item.show &&
```
Comment (idlefingers) [inline code/src/components/TodoItem.js]: Same here with the `return null` trick

File: code/src/pages/Home.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import { todos } from '../reducers/todos'
+import { TodoList } from '../components/TodoList'
+
+// Variable combining reducers from todos
+const reducer = combineReducers({ todos: todos.reducer })
+
+// Persistence: Retrieves the existing state from localstorage if it exists
+const persistedState = localStorage.getItem('reduxState')
+  ? JSON.parse(localStorage.getItem('reduxState'))
+  : {};
```
Comment (idlefingers) [inline code/src/pages/Home.js:13]: Nice!

**Review by idlefingers — APPROVED** [review]
Amazing work Frida! Super nice & clean code, makes it a pleasure to read through. Looks like you have a great grasp of both react and redux to me. Awesome! ⭐

File: code/src/components/Filters.js (hunk 3 above)
Comment (fridamaria) [reply code/src/components/Filters.js]: Ah, sounds logical! Will try it out :)

File: code/src/components/Filters.js (hunk 4 above)
Comment (fridamaria) [reply code/src/components/Filters.js]: Makes sense, I'll remove it!

File: code/src/components/TodoInput.js (hunk 5 above)
Comment (fridamaria) [reply code/src/components/TodoInput.js]: Ah, I missed that one :)

File: code/src/components/TodoItem.js (hunk 6 above)
Comment (fridamaria) [reply code/src/components/TodoItem.js]: Done :)

**Pull request closed without merging**